### ISSUE TITLE
UpdateProjectLevelRuleOptions should not omit empty GroupIDs/UserIDs

### ISSUE THREAD

**Issue by QuingKhaos** [xanzy/go-gitlab#1292]
### SDK Version

```
// go.mod
github.com/xanzy/go-gitlab v0.51.1
```

```
// go.sum
github.com/xanzy/go-gitlab v0.51.1 h1:wWKLalwx4omxFoHh3PLs9zDgAD4GXDP/uoxwMRCSiWM=
github.com/xanzy/go-gitlab v0.51.1/go.mod h1:Q+hQhV508bDPoBijv7YjK/Lvlb4PhVhJdKqXVQrUoAE=
```

### Expected Behavior

When calling `client.Projects.UpdateProjectApprovalRule` with empty `GroupIDs` or empty `UserIDs` it should send an empty attribute to the API and thus removing all groups/users from the project-level merge request approval rule.

### Actual Behavior

Due to the `omitempty` attribute in the `UpdateProjectLevelRuleOptions` struct, an empty group/user list is not send in the PUT request to the API and I'm not able to remove all groups or users while updating the project-level merge request approval rule.

### References

- #700 
- #717 
- gitlabhq/terraform-provider-gitlab#755 - the issue in the GitLab Terraform Provider raising this issue

**Comment by svanharmelen** [comment]
PR #1294 is a nasty breaking change, but it does fix this issue...

**Comment by QuingKhaos** [comment]
Nasty 😅 

I don't know Go well enough, but with a pointer `omitempty` then omits it only, when the pointer is empty and not the list?

With a BC like that, that will take the GitLab Terraform Provider a while to catch up ^^

**Comment by svanharmelen** [comment]
Yeah correct about the behavior of `omitempty` with a pointer... And it's really simple and straight forward to fix, so I would be surprised if it would take a while for the TF provider to update to the latest version.

**Comment by QuingKhaos** [comment]
Straightforward, but still needs time and a new release ^^

**Comment by svanharmelen** [comment]
This should be fixed by PR #1294 which is just merged...
